### PR TITLE
[test] Do not fail HooksTestStaticInit test in IDE

### PR DESCRIPTION
**Pull request by szpak**
Playing with the Reactor code I have noticed that `HooksTestStaticInit` fails when a group of tests is executed from IDE. After reading the code the reason is obvious, however I understand why it was created that way.

It's a corner case test that is separately executed from Gradle. After my change, when executed "accidentally" from IDE, it is skipped - to do not have test execution red. There is a dedicated check in Gradle to ensure that test is not skipped accidentally by some configuration changes.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@szpak Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=1322)!

[Click here](https://cla.pivotal.io/sync/reactor?repositoryId=reactor/reactor-core&pullRequestId=1322) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by szpak** [comment]
I believe I have agreed to the CLA long time ago...

**Comment by pivotal-issuemaster** [comment]
@szpak Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=1322)!

**Comment by simonbasle** [comment]
Hey @szpak !

 - What IDE did you use?
 - What group of tests did you try to run?

I never saw this problem in IntelliJ using either `gradle check`, `gradle test` or manually executing one to several test classes.

**Comment by szpak** [comment]
@simonbasle Just run all tests for package `reactor` in `reactor-core` using IntelliJ IDEA (tested with 2018.2, but it shouldn't matter). Idea detects a test in `HooksTestStaticInit` (even though there is no a `Test` suffix), runs it and there is a failure (due to `reactor.trace.operatorStacktrace` no being set).
It's a minor issue (due to a number of tests people usually don't run all the tests), but as I don't like failing tests and it was easy to fix (workaround) I proposed an enhancement.

File: build.gradle [hunk 1]
```diff
@@ -377,6 +377,11 @@ project('reactor-core') {
 	doFirst {
 	  println "Additional tests from `testStaticInit` ($includes)"
 	}
+	afterSuite { TestDescriptor descriptor, TestResult result ->
```
Comment (simonbasle) [inline build.gradle]: the `afterSuite` only applies to this particular test task, correct? (ie. if I exclude the testStaticInit task from the gradle command line, it won't cause any issue)

File: reactor-core/src/test/java/reactor/core/publisher/HooksTestStaticInit.java [hunk 2]
```diff
@@ -36,6 +36,8 @@
 import org.junit.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
```
Comment (simonbasle) [inline reactor-core/src/test/java/reactor/core/publisher/HooksTestStaticInit.java]: could you rewrite to use pure `AssertJ` instead? it as an `Assumptions` class that you can use for assumptions.

File: build.gradle (hunk 1 above)
Comment (szpak) [reply build.gradle]: Yes, `afterSuite` is per the `Test` task instance.

File: reactor-core/src/test/java/reactor/core/publisher/HooksTestStaticInit.java (hunk 2 above)
Comment (szpak) [reply reactor-core/src/test/java/reactor/core/publisher/HooksTestStaticInit.java]: Done. In fact as a die-hard Spock user I haven't used `Assumptions` from AssertJ before, but it looks ok (of course not so nice as `@Require` or `@IgnoreIf` in Spock ;) ).

**Comment by simonbasle** [comment]
Thanks for the PR 🎁@szpak!
